feat: Add a GH action to create a patch release branch

RomanDavydchuk opened the pull request:
## Summary

Add a GH action to create a patch release branch. This action should be called by the "Create Patch Release PR" n8n workflow on internal instance in order to automate parts of the patch release. The flow should look something like this:
1. Someone triggers the n8n workflow, specifying the old version, new version and optionally extra PRs to be included in the patch release
2. The workflow automatically finds the PRs with label `release/backport` and without `Released` label. The workflow gets the merge commit SHAs of those PRs and the extra PRs specified before
3. The workflow triggers this GH action, which creates and pushes the `patch/1.XXX.X` branch by cherry-picking the commits that are found by the n8n workflow
4. n8n workflow waits for this action to complete
5. n8n workflow triggers the GH action to create a release PR using the branch that was created by this GH action

This should eliminate the need to manually create the branch, cherry-pick the commits and push the branch to the repo (given there are no conflicts)

If there are conflicts, the GH action to create a release PR will error (since there would be no `patch/1.XXX.X` branch), in this case we need to manually cherry-pick the commits and resolve the conflicts, nothing can be done here

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
https://linear.app/n8n/issue/
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->


## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

RomanDavydchuk committed:
feat: Add GH action

tomi commented on .github/workflows/create-patch-release-branch.yaml:
```diff
@@ -0,0 +1,44 @@
+        with:
+          fetch-depth: 0
+      - name: Setup, cherry-pick and push branch
+        continue-on-error: true
```
Why do we continue on error? Are we handling the error somewhere?

tomi commented on .github/workflows/create-patch-release-branch.yaml:
```diff
@@ -0,0 +1,44 @@
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup, cherry-pick and push branch
```
Would be good to verify that the new and old version adhere to proper semver:

```
- name: Validate inputs
    run: |
      # Make sure the versions adhere to semver, e.g. 1.110.0 and 1.110.0-exp.1
      if ! [[ "${{ inputs.old_version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z]+(\.[0-9A-Za-z]+)*)?$ ]]; then
        echo "Invalid old version format: ${{ inputs.old_version }}"
        exit 1
      fi
# ... and new_version
```

RomanDavydchuk committed:
feat: Add input validation, better error handling

RomanDavydchuk commented:
Pushed some changes
- added input validation to make sure old and new versions are following SemVer format
- added a check to skip commits that are already in the branch
- force pushing the patch branch in a case we need to retry running the action
- improved error handling by sending a proper response to the n8n workflow with a message in case something went wrong
I did update the n8n workflow as well to support these changes and added some minor improvements too

tomi reviewed (approved):
Nice job 👏 Thank you for implementing this 💟
